dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureIntrospector.java

```diff
@@ -0,0 +1,113 @@
+     */
+    @Override
+    public void introspect(PrintWriter out) throws Exception {
+        Set<String> features = this.provisioner.getInstalledFeatures();
+        for (String feature : features) {
+            FeatureDefinition featureDef = this.provisioner.getFeatureDefinition(feature);
+            out.println(featureDef.getFeatureName());
+            out.println("    Visibility: " + featureDef.getVisibility());
+            out.println("    Version: " + featureDef.getVersion());
+            out.println("    Simple Name: " + featureDef.getFeatureName());
+            out.println("    Symbolic Name: " + featureDef.getSymbolicName());
+            if (featureDef instanceof ProvisioningFeatureDefinition) {
+
+                ProvisioningFeatureDefinition proFeatDef = (ProvisioningFeatureDefinition) featureDef;
+
+                String bundleRepType = proFeatDef.getBundleRepositoryType();
+                out.println("    Bundle Repository Type: " + (bundleRepType == null || bundleRepType.isEmpty() ? "core" : bundleRepType));
+                out.println("    Auto Feature: " + proFeatDef.isAutoFeature());
+                out.println("    Singleton: " + proFeatDef.isSingleton());
+
+                if (proFeatDef instanceof SubsystemFeatureDefinitionImpl) {
+                    SubsystemFeatureDefinitionImpl subSysFeatureDef = (SubsystemFeatureDefinitionImpl) proFeatDef;
+                    ProvisioningDetails details = subSysFeatureDef.getProvisioningDetails();
+                    boolean detailsAreSet = false;
+                    //Create provisioning details if not already set
+                    if (details == null) {
+                        try {
+                            details = new ProvisioningDetails(subSysFeatureDef.getImmutableAttributes().featureFile, null);
+                            details.setImmutableAttributes(subSysFeatureDef.getImmutableAttributes());
+                            subSysFeatureDef.setProvisioningDetails(details);
+                            detailsAreSet = true;
+                        } catch (IOException e) {
+                            //AutoFFDC is fine here
+                        }
+                    }
+                    boolean isSuperseded = details.isSuperseded();
+                    out.println("    Superseded: " + isSuperseded);
+                    if (isSuperseded) {
+                        out.println("        Superseded by: " + details.getSupersededBy());
+                    }
+                    Collection<FeatureResource> constituents = details.getConstituents(null);
+                    out.println("    Constituents: ");
+                    for (FeatureResource constituent : constituents) {
+                        out.println("        " + constituent.getSymbolicName());
+                        out.println("            " + constituent.getAttributes().toString());
+                    }
+                    if (detailsAreSet) {
```

Comment:
Just after the initialization of `detailsAreSet` at line 82 I would add a try-finally block.  Then put this `if (detailsAreSet) {}` code inside the `finally` block.